src/goto-instrument/synthesizer/synthesizer_utils.h

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************\
+
+Module: Utility functions for loop invariant synthesizer.
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_SYNTHESIZER_UTILS_H
+#define CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_SYNTHESIZER_UTILS_H
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/goto_program.h>
+
+#include <analyses/loop_analysis.h>
+#include <goto-instrument/loop_utils.h>
```

Comment:
This is confusing. There's a loop_utils.h, then loop_idt is in synthesizer_utils.h, there's also loop_ids.h elsewhere in the codebase. Declaring things in appropriately named header files increases discoverability.

Reply:
Good point. I moved the definition of `loop_idt` to `loop_ids.h`.